vue-hocr/src/components/HocrViewer/feature/ScaleFont.js

```diff
@@ -4,53 +4,46 @@ export default class ScaleFont {
 
   constructor({fontFamily='x', fonts={}}) {
     this.fontFamily = fontFamily
-    this.maxFontSize = 128
     this.minFontSize = 2
     this.wrapClass   = 'hocr-viewer-wrap'
     this.cache = {}
+    // wrapper element containing wrappers for font-size expansion
+    this.wrap = document.createElement('span')
+    this.wrap.classList.add(this.wrapClass)
   }
 
   apply(dom) {
     console.time('toggleScaleFont')
 
-    // wrapper element containing wrappers for font-size expansion
-    const wrap = document.createElement('span')
-    wrap.classList.add(this.wrapClass)
-    document.body.appendChild(wrap)
-    HocrDOM.queryHocrAll(dom, {terminal: true}).forEach((el) => this.scaleFont(el, wrap))
-    wrap.remove()
+    document.body.appendChild(this.wrap)
+    HocrDOM.queryHocrAll(dom, {terminal: true}).forEach((el) => this.scaleFont(el))
 
     console.timeEnd('toggleScaleFont')
   }
 
-  scaleFont(el, wrap) {
+  scaleFont(el) {
     const text = el.textContent.trim()
     if (text.length === 0) return
-    if (!(text in this.cache)) {
-      // wrap.setAttribute('class', el.getAttribute('class'))
-      // wrap.style.width = '100%'
-      wrap.style.fontFamily = el.style.fontFamily
-      wrap.innerHTML = text
-      const height = parseInt(el.style.height.replace('px', ''))
-      const width = parseInt(el.style.width.replace('px', ''))
-      let fontSize = Math.min(width, height)
-      if (fontSize <= this.minFontSize) {
-        this.cache[text] = this.minFontSize
-      } else {
-        // console.log({fontSize, width: width, height:height, min, el})
-        wrap.style.fontSize = fontSize + 'px'
-        if (fontSize > this.minFontSize && wrap.offsetHeight > height) {
-          fontSize -= wrap.offsetHeight - height
-          wrap.style.fontSize = fontSize + 'px'
-        }
-        while (fontSize > this.minFontSize && wrap.offsetWidth > width) {
-          wrap.style.fontSize = --fontSize + 'px'
-        }
-        // if (iterations > 1) console.debug(iterations, text, wrap.offsetHeight, height, wrap.offsetWidth, width)
-        this.cache[text] = fontSize - 1
+    this.wrap.style.fontFamily = el.style.fontFamily
+    this.wrap.innerHTML = text
+    const height = parseInt(el.style.height.replace('px', ''))
+    const width = parseInt(el.style.width.replace('px', ''))
+    let fontSize = height
+    if (fontSize <= this.minFontSize) {
+      fontSize = this.minFontSize
+    } else {
+      // console.log({fontSize, width: width, height:height, min, el})
+      this.wrap.style.fontSize = fontSize + 'px'
+      if (fontSize > this.minFontSize && this.wrap.offsetHeight > height) {
```

Comment:
Using similar code (`if (this.wrap.offsetHeight + DELTA < height)`) we could increase the font size if it does not fill the box height. That would also fix the problem with text without ascenders and descenders.